src/erlcloud_cur.erl

```diff
@@ -0,0 +1,133 @@
+-module(erlcloud_cur).
+
+
+%% AWS Cost and Usage Report API implementation
+%% https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/api-reference.html
+
+-include("erlcloud_aws.hrl").
+
+%% Initialization API
+-export([new/2, new/3, new/4, new/5]).
+
+%% API
+-export([
+    describe_report_definitions/1,
+    describe_report_definitions/2
+]).
+
+
+%%------------------------------------------------------------------------------
+%% Initialization API
+%%------------------------------------------------------------------------------
+
+-spec new(string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec new(string(), string(), string()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                cur_host          = Host,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec new(string(), string(), string(), non_neg_integer()) -> aws_config().
+new(AccessKeyID, SecretAccessKey, Host, Port) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                cur_host          = Host,
+                cur_port          = Port,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+-spec new(string(), string(), string(), non_neg_integer(), string()) ->
+    aws_config().
+new(AccessKeyID, SecretAccessKey, Host, Port, Scheme) ->
+    #aws_config{access_key_id     = AccessKeyID,
+                secret_access_key = SecretAccessKey,
+                cur_host          = Host,
+                cur_port          = Port,
+                cur_scheme        = Scheme,
+                retry             = fun erlcloud_retry:default_retry/1}.
+
+
+%%------------------------------------------------------------------------------
+%% API
+%%------------------------------------------------------------------------------
+
+%% @doc
+%% Cost & Usage Report API:
+%% https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/describe-report-definitions.html
+%%
+%% `
+%% {ok, R} = erlcloud_cur:describe_report_definitions(Config).
+%% `
+%%
+-spec describe_report_definitions(aws_config()) -> {ok, map()} | {error, any()}.
+describe_report_definitions(Config) ->
+    describe_report_definitions(#{}, Config).
+
+
+%% `
+%% Opts = #{<<"MaxResults">> => 5},
```

Comment:
from https://docs.aws.amazon.com/awsaccountbilling/latest/aboutv2/describe-report-definitions.html:

```
MaxResults
Optional integer. Specifies the maximum number of results to return in response.
Type: Integer
Valid value: 5
Required: No
```

* so it's possible only to have it equal 5, and
* also it looks like 5 is maximum number of reports which is possible (the button becomes inactive in UI when trying to create > 5)
* I haven't found documentation on whether there is `next_token`, so I left this part of retrieving all objects not implemented (although it's already possible to submit token as part of options)

Reply:
Actually, `aws cur describe-report-definitions help` says:

```
NextToken -> (string)
          A generic string.
```

so I can implement describe_report_definitions_all which would retrieve all reports, and this would work once in future it changed (provided they really will use `NextToken` as I don't how to test it now)

Reply:
Let's keep it as is and have `describe_report_definitions_all` implemented once the doc is updated with `NextToken` information.

Reply:
token is to be passed as `Options`. no more *_all funs